.github/workflows/testing.yml

```diff
@@ -0,0 +1,18 @@
+name: Go unit tests
+
+on: [push, workflow_dispatch]
+
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
Is `fetch-depth` necessary here?

Reply:
We only want to test the last commit, and we don't really need the entire Git history (unless you're releasing and you need things like tags or branches).

Were you thinking to add something like that in the future that would warrant the tests to have access to the entire Git history?

Reply:
If we don't need it for the current workflow, I'd suggest not using it until it is actually required. It's probably not the most problematic thing, but more my general philosophy.

Reply:
Sorry I'm lost, are you saying "not use the commit history until it's actually required"? (I'm good with that, that's what the depth set to zero does!)

If it's around "not using the `fetch-depth` feature until it's actually required"... The only time you would use it is in this scenario, where you really care about the last commit only, like here: we just want to test the Go code and run unit tests, not whether 30 commits before you needed to run those tests too!). 

Other situations definitely don't need the `fetch-depth` (like, for example, [the release flow](https://github.com/kubefirst/kubefirst-api/blob/main/.github/workflows/release.yaml) since it needs the git tags and other pieces of the git ref log).

We can always come back to it if need be though!